File: backend/src/config/env.js
import dotenv from 'dotenv';

dotenv.config();

const requiredMongoUri = process.env.MONGODB_URI;

export const env = {
  port: process.env.PORT || 4000,
  mongodbUri: requiredMongoUri,
  mediaBaseUrl: process.env.MEDIA_BASE_URL || 'http://localhost:4000',
};

if (!env.mongodbUri) {
  throw new Error('MONGODB_URI is required. Set it in Vercel environment variables.');
}

File: lib/core/config/env.dart
import 'package:flutter_dotenv/flutter_dotenv.dart';

class Env {
  static String apiBaseUrl() {
    return dotenv.env['API_BASE_URL'] ?? 'http://localhost:4000';
  }

  static String mediaBaseUrl() {
    return dotenv.env['MEDIA_BASE_URL'] ?? apiBaseUrl();
  }
}
